### PR TITLE
Add Modal component

### PR DESCRIPTION
**Pull request by felixjung**
##### Changes

- Adds support for media queries
- Adds a Modal implementation. Right now this requires a provider to sit at the top of the tree. I kept that implementation because in the past there was a requirement which forced us to use it like this. If we can, I would look to go to a simpler implementation.
- Bug fixes and improvements
  - Fixes `CardHeader` layout when no title is provided but a closing button is required.
  - Removes margin from `Heading` in a Card story.
  - Adds `noMargin` prop to blacklist of `HtmlElement` in `Heading` component.
  - Adds `primary` prop to blacklist of `HtmlElement` in `Button` component.
  - Removes a broken export of a `sizes` module from `utils`.

##### Usage

- `ModalProvider` has to sit on top of the component tree.
- Consumer, exported as `Modal` from the `Modal/index.js`, provides a `set` and a `get` function to its render prop.
- create modal by passing a modal config with the props `children` and `onClose` to the `set` function. 
- Modal already comes with `Card` as wrapper.
- `children` render prop is used as content of the card.
- Provide title via prop.
- Provide buttons via `buttons` render prop. Use `ButtonGroup` for this.

```javascript
      const modal = {
        onClose: console.log,
        title: 'My cool modal',
        children: () => (
            <Text>Some text in the modal body.</Text>
        ),
        buttons: ({ onClose }) => (
          <ButtonGroup>
            <Button key="confirm" onClick={onClose} data-test="confirm-button">
              Confirm Button
            </Button>
            <Button key="cancel" onClick={onClose} data-test="cancel-button">
              Cancel Button
            </Button>
          </ButtonGroup>
        )
      };
      return <PageWithModal modal={modalWithTitle} />;
    }

  <ModalProvider>
    <Modal>
      {({ setModal }) => (
        <Button
          type="button"
          onClick={() => {
            setModal(modal);
          }}
        >
          Open modal
        </Button>
      )}
    </Modal>
  </ModalProvider>
```

###### Desktop
desktop-modals

###### Mobile
mobile-modals

### CODE REVIEW

File: src/components/Modal/Modal.spec.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    expect(wrapper.find('Card')).toHaveLength(1);
+  });
+
+  it('should close', async () => {
+    const wrapper = openModal(defaultModal);
+    const closeButton = wrapper.find('SvgButton').find('button');
+    closeButton.simulate('click');
+    await new Promise(resolve => {
+      setTimeout(() => resolve());
+    });
```
Comment (connor-baer) [inline src/components/Modal/Modal.spec.js]: Why is this timeout necessary?

File: src/components/Modal/ModalProvider.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+export class ModalProvider extends Component {
+  static propTypes = {
+    children: childrenPropType.isRequired
+  };
+
+  state = {
+    modal: null,
+    isOpen: false
+  };
```
Comment (connor-baer) [inline src/components/Modal/ModalProvider.js]: Is this a new syntax to set the initial state?

Comment (felixjung) [reply src/components/Modal/ModalProvider.js]: Yeah, this is [property initializer syntax](https://www.fullstackreact.com/articles/use-property-initializers-for-cleaner-react-components/). I believe you've used this in a bunch of places yourself already, no? I know I have. Works just like when declaring a method on a class component.

Just one of those future JS features that got picked up early by the React community.

Comment (connor-baer) [reply src/components/Modal/ModalProvider.js]: I had seen it before but wasn't sure how/when to use it. The article you linked clears it up. Thanks! :)

**Comment by felixjung** [comment]
Shall we merge this? I kind of need it now and it would also unblock @Vilsonacneto.

**Comment by connor-baer** [comment]
Let's merge!

**Pull request merged**